insights/specs.py

```diff
@@ -424,7 +425,23 @@ def block(broker):
 tomcat_web_xml = sf.first_of([sf.glob_file("/etc/tomcat*/web.xml", name="tomcat_web_xml_etc"),
                               sf.glob_file("/conf/tomcat/tomcat*/web.xml", name="tomcat_web_xml_conf")],
                               name="tomcat_web_xml", alias="tomcat_web.xml")
-tomcat_virtual_dir_context = sf.simple_command("/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*", name="tomcat_virtual_dir_context")
+
+
+@datasource(HostContext)
+def tomcat_home(broker):
+    ctx = broker[HostContext]
+    ps = ctx.shell_out("ps auxwww")
+    results = []
+    findall = re.compile(r"\-Dcatalina\.(home|base)=(\S+)").findall
+    for p in ps:
+        found = findall(p)
+        if found:
+            results.extend(f[1] for f in found)
+    return list(set(results))
+
+
+tomcat_vdc_targeted = sf.foreach(tomcat_home, "/bin/grep -R -s 'VirtualDirContext' --include '*.xml' %s", name="tomcat_vdc_targeted")
+tomcat_vdc_fallback = sf.simple_command("/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*", name="tomcat_vdc_fallback")
```

Comment:
In 1.x, `tomcat_vdc_fallback` corresponding command is `"/usr/bin/find /usr/share -maxdepth 1 -name 'tomcat*' -exec /bin/grep -R -s 'VirtualDirContext' --include '*.xml' '{}' +"`, in this PR, it's `/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*`, though these commands work the same, do you think it's better to keep the same command format?

Reply:
@zhangyi733 Good catch. They are not the same and the one with glob (the old one) does not work.